lib/asciidoctor/converter/html5.rb

```diff
@@ -1257,13 +1257,17 @@ def convert_inline_quoted node
 
   # NOTE expose read_svg_contents for Bespoke converter
   def read_svg_contents node, target
-    if (svg = node.read_contents target, start: (node.document.attr 'imagesdir'), normalize: true, label: 'SVG')
+    if (svg = node.read_contents target, start: (node.document.attr 'imagesdir'), normalize: true, label: 'SVG', warn_if_empty: true)
+      return if svg.empty?
       svg = svg.sub SvgPreambleRx, '' unless svg.start_with? '<svg'
-      old_start_tag = new_start_tag = nil
+      old_start_tag = new_start_tag = start_tag_match = nil
       # NOTE width, height and style attributes are removed if either width or height is specified
       ['width', 'height'].each do |dim|
         if node.attr? dim
-          new_start_tag = (old_start_tag = (svg.match SvgStartTagRx)[0]).gsub DimensionAttributeRx, '' unless new_start_tag
+          unless new_start_tag
+            next if (start_tag_match ||= (start_tag_match == false ? false : (start_tag_match = (svg.match SvgStartTagRx) || false))) == false
```

Comment:
It can be simplified no?
The inner ternary condition `start_tag_match == false ? ...` is always false because we already check if `start_tag_match` equals `false` in the outer condition (and do `next` if that is the case).
So in the inner condition `start_tag_match` is always not `false`.

Reply:
I think the inner condition (i.e. right assignment of `||=`) will be evaluated because `start_tag_match` is false...

Reply:
But we can probably remove one assignment since we are using `start_tag_match ||=`:

```suggestion
            next if (start_tag_match ||= (start_tag_match == false ? false : ((svg.match SvgStartTagRx) || false))) == false
```

Arguably, we should check if `start_tag_match` is `nil` to avoid the extra `start_tag_match == false ? false : ...`.

Not sure what is the one-line syntax but basically what we are doing here is:

```rb
if start_tag_match.nil?
  start_tag_match = (svg.match SvgStartTagRx) || false
end
next if start_tag_match == false
```

Reply:
The `||=` operator is nice but since we need to make a distinction between `nil` and `false` values, the code becomes less obvious.

Reply:
This code optimizes for the virtuous path that the start tag match is found on the previous run. When it is not set, we have to check if it's false to indicate it failed on a previous attempt. The end result is that it only ever runs ones. It may not be beautiful, but it does work.

Reply:
If we were to rewrite it, I would go with something like:

```
next if (start_tag_match ||= (svg.match SvgStartTagRx) || :no_match) == :no_match
```

That way we can leverage the `||=` operator effectively. And it seems clear what is happening.

Reply:
Are you saying that the following code:

```rb
start_tag_match = ((svg.match SvgStartTagRx) || false) if start_tag_match.nil?
next if start_tag_match == false
```

does more work than:

```rb
next if (start_tag_match ||= (start_tag_match == false ? false : (start_tag_match = (svg.match SvgStartTagRx) || false))) == false
```

As far as I understand Ruby still has to evaluate the value of `start_tag_match` twice right? 
1. `start_tag_match` is trueish -> don't do the assignment (`||=` operation)
2. `start_tag_match` is not equals to `false` -> don't execute `next` (`==` operation)

I guess I could try to a do micro benchmark but the array is only two elements long so the difference will be on the nanosecond range :laughing:

Reply:
I rewrote it. See if you like the new way.

Reply:
That's perfect, the code is concise and the intention is clear :1st_place_medal: